source/util/ilist_node.h

```diff
@@ -120,15 +185,46 @@ inline void IntrusiveNodeBase<NodeType>::InsertAfter(NodeType* pos) {
 template <class NodeType>
 inline void IntrusiveNodeBase<NodeType>::RemoveFromList() {
   assert(!this->is_sentinel_ && "Sentinel nodes cannot be moved around.");
-  assert(this->next_node_ != nullptr && "Cannot remove a node from a list if it is not in a list.");
-  assert(this->previous_node_ != nullptr && "Cannot remove a node from a list if it is not in a list.");
+  assert(this->IsInAList() &&
+      "Cannot remove a node from a list if it is not in a list.");
 
   this->next_node_->previous_node_ = this->previous_node_;
   this->previous_node_->next_node_ = this->next_node_;
   this->next_node_ = nullptr;
   this->previous_node_ = nullptr;
 }
 
+template<class NodeType>
+void IntrusiveNodeBase<NodeType>::ReplaceWith(NodeType* target) {
+  if (is_sentinel_) {
```

Comment:
I don't understand the logic in this if/else. Please check if you are not confusing 'this' with 'target'.

Reply:
Yes, there were some problems there.  I fixed it up, please take another look when I do a push.